airgun/views/ansible_role.py

```diff
@@ -0,0 +1,72 @@
+from widgetastic.widget import Checkbox
+from widgetastic.widget import Table
+from widgetastic.widget import Text
+from widgetastic_patternfly import BreadCrumb
+from widgetastic_patternfly4 import Button
+from widgetastic_patternfly4 import PatternflyTable
+
+from airgun.views.common import BaseLoggedInView
+from airgun.views.common import SearchableViewMixin
+from airgun.widgets import ActionsDropdown
+from airgun.widgets import Pagination
+
+
+class ImportPagination(Pagination):
+    PER_PAGE_BUTTON_DROPDOWN = ".//div[button[@id='pagination-options-menu-toggle-2']]"
+
+
+class AnsibleRolesView(BaseLoggedInView, SearchableViewMixin):
+    """Main Ansible Roles view. Prior to importing any roles, only the import_button
+    is present, without the search widget or table.
+    """
+
+    title = Text("//h1[contains(., text()='Ansible Roles')")
+    import_button = Text("//a[contains(@href, '/ansible_roles/import')]")
+    submit = Button('Submit')
+    total_imported_roles = Text("//span[contains(@class, 'pagination-pf-items-total')]")
+    table = Table(
+        './/table',
+        column_widgets={
+            '▲ Name	': Text("./a"),
```

Comment:
I think I would raise this as an issue with the UX team, for testability. They have a gchat channel we can engage on instead of going straight to writing a BZ.

Reply:
In the meantime try using `widgetastic.utils.partial_match` instance for the key here, like `partial_match('Name')`. I'm not confident it will work correctly as a dictionary key, but its worth a try.

Reply:
Thanks for pointing me to the UX team's channel. I will try the `partial_match` util as well and update with the results here.

Reply:
I could not figure out how to get `partial_match` working as a dictionary key, but I came up with another workaround: as long as we're searching for a specific role to delete, there will only be one row in the search results, so `view.table.row()['Actions'].widget.fill('Delete')` works. This is not optimal but I believe it will suffice until the underlying behavior changes (i.e. the UX team decides to either not sort roles by name on import or wraps the sort character in a separate `span` tag).